src/go/rpk/pkg/cli/cmd/common/common.go

```diff
@@ -159,14 +167,38 @@ func DeduceBrokers(
 }
 
 func CreateProducer(
-	brokers func() []string, configuration func() (*config.Config, error),
+	brokers func() []string,
+	configuration func() (*config.Config, error),
+	tlsConfig func() (*config.TLS, error),
+	authConfig func() (*config.SCRAM, error),
 ) func(bool, int32) (sarama.SyncProducer, error) {
 	return func(jvmPartitioner bool, partition int32) (sarama.SyncProducer, error) {
 		conf, err := configuration()
 		if err != nil {
 			return nil, err
 		}
-		cfg, err := kafka.LoadConfig(conf)
+
+		tls, err := tlsConfig()
+		if err != nil {
+			return nil, err
+		}
+		// If no TLS config was set, try to look for TLS config in the
+		// config file.
+		if tls == nil {
+			tls = &conf.Rpk.TLS
+		}
+
+		scram, err := authConfig()
+		if err != nil {
+			return nil, err
+		}
+		// If no SCRAM config was set, try to look for it in the
+		// config file.
+		if scram == nil {
+			scram = &conf.Rpk.SCRAM
+		}
```

Comment:
Does the `authConfig()` returns `ErrNoCredentials`?
```suggestion
		scram, err := authConfig()
		if err != nil {
			return nil, err
		}
		// If no SCRAM config was set, try to look for it in the
		// config file.
		if errors.Is(err, ErrNoCredentials) {
			scram = &conf.Rpk.SCRAM
		}
```

Reply:
Thanks for this. Good catch.